internal/app/machined/pkg/runtime/v1alpha1/platform/equinixmetal/equinix.go

```diff
@@ -369,9 +369,17 @@ func (p *EquinixMetal) FireEvent(ctx context.Context, event Event) error {
 		retry.WithErrorLogging(true)).RetryWithContext(
 		ctx,
 		func(ctx context.Context) error {
-			_, err = http.Post(*eventURL, "application/json", bytes.NewBuffer(eventData))
+			req, reqErr := http.NewRequestWithContext(ctx, http.MethodPost, *eventURL, bytes.NewReader(eventData))
+			if reqErr != nil {
+				return reqErr
+			}
+
+			body, reqErr := http.DefaultClient.Do(req)
+			if body != nil {
+				body.Body.Close() //nolint:errcheck
```

Comment:
I think it also needs to read out the body to prevent HTTP requests blocking the connection:

```go
io.Copy(io.Discard, resp.Body)
```